sgx-isa/src/lib.rs

```diff
@@ -694,3 +697,77 @@ fn test_eq() {
     b.miscmask = 0xdeadbeef;
     assert!(a == b);
 }
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+use core::mem::MaybeUninit;
+
+/// Wrapper struct to force 16-byte alignment.
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+#[repr(align(16))]
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub struct Align16<T>(pub T);
+
+/// Wrapper struct to force 128-byte alignment.
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+#[repr(align(128))]
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub struct Align128<T>(pub T);
+
+/// Wrapper struct to force 512-byte alignment.
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+#[repr(align(512))]
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub struct Align512<T>(pub T);
+
+const ENCLU_EREPORT: u32 = 0;
+const ENCLU_EGETKEY: u32 = 1;
+
+/// Call the `EGETKEY` instruction to obtain a 128-bit secret key.
+#[cfg(all(feature = "nightly", target_env = "sgx"))]
+#[unstable(feature = "sgx_platform", issue = "56975")]
+pub fn egetkey(request: &Align512<[u8; 512]>) -> Result<Align16<[u8; 16]>, u32> {
```

Comment:
These shouldn't be exposed as free functions, instead, these should be exposed the same way the `std::os::fortanix_sgx` functions are used now (see elsewhere in this file)

Reply:
Should it be in a different file in the same project in rust-sgx repo, or should it be in the rustlang repo?

Reply:
It should be in this file, or perhaps in a submodule.